share/getters/shrex.go

```diff
@@ -268,3 +290,14 @@ func (sg *ShrexGetter) GetSharesByNamespace(
 			"finished (s)", time.Since(reqStart))
 	}
 }
+
+func (sg *ShrexGetter) getPeer(
+	ctx context.Context,
+	header *header.ExtendedHeader,
+) (libpeer.ID, peers.DoneFunc, error) {
+	if sg.archivalPeerManager != nil && !pruner.IsWithinAvailabilityWindow(header.Time(), sg.availabilityWindow) {
+		p, df, err := sg.archivalPeerManager.Peer(ctx, header.DAH.Hash(), header.Height())
+		return p, df, err
+	}
+	return sg.fullPeerManager.Peer(ctx, header.DAH.Hash(), header.Height())
```

Comment:
Also if sg.availabilityWindow == 0, it will always result in call of `pruner.IsWithinAvailabilityWindow` to return true, even for historical. This default works well for DASer, but is incorrect here. This default setting will result in all request to be routed to prunned nodes. Perhaps lets disable sg.availabilityWindow == 0 being possible on params validation at least, to ensure misconfiguration does not happen.

Reply:
@walldiss full peers are not necessarily pruned 

archival peers also advertise on `full` topic.

Reply:
I still don't understand why the logic here isn't applicable. `full` topic contains *all* FN + BN in the network, not just pruned, so why would routing peer requests to `full` peers be an issue? It's how the network worked before as well. If it's just semantics, then maybe we can make an issue for it?

Reply:
Because sg.availabilityWindow == 0 disables routing to prunned nodes. You never want to disable routing here, regardless of local availability window setup. 
Perhaps confusion comes from the fact that it is configurable in the first place, when it should be just a constant.